include/mbgl/style/expression/in.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <mbgl/style/conversion.hpp>
+#include <mbgl/style/expression/expression.hpp>
+#include <memory>
+
+namespace mbgl {
+namespace style {
+namespace expression {
+
+class In : public Expression {
+public:
+    In(std::unique_ptr<Expression> needle_, std::unique_ptr<Expression> haystack_)
+        : Expression(Kind::In, type::Boolean), needle(std::move(needle_)), haystack(std::move(haystack_)) {}
```

Comment:
could you add asserts for needle and haystack

Reply:
We have checks in `evaluate` and `parse` method. If add asserts here it will be duplicate.

Reply:
When someone would add `in` expression to dsl, constructor might be called directly, without parse. So, would be good to have guards in constructor

Reply:
ok